prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -188,13 +188,15 @@ def generate_final_decision(
             verbose=2,
         )
 
+        logger.info(f"Starting to generate final decision for '{question}'.")
         crew.kickoff(
             inputs={
-                "scenarios_with_probabilities": [
-                    (i[0], i[1].model_dump()) for i in scenarios_with_probabilities
-                ],
+                "scenarios_with_probabilities": "\n".join(
+                    f"- Scenario '{s}' has probability of happening {a.p_yes * 100:.2f}% with confidence {a.confidence * 100:.2f}%, because {a.reasoning}"
+                    for s, a in scenarios_with_probabilities
+                ),
                 "number_of_scenarios": len(scenarios_with_probabilities),
-                "scenario_to_assess": scenarios_with_probabilities[0][0],
+                "scenario_to_assess": question,
```

Comment:
Scenarios could be generated out of order, and then the `scenario_to_assess` could be different from the user's question.